### PR TITLE
PassVsprintfValidator: Add separate regex for arguments only

### LINKED ISSUE #16421: Cannot add literal % to translation strings
**Describe the bug**
Any translation with a literal percent `%%` will fail because it is not an argument but is counted as one in `PrestaShopTranslatorTrait::$regexSprintfParams` and `PassVsprintfValidator::countArgumentsOfTranslation()`

**To Reproduce**
1. Go to any translatable string
2. Add a literal percent `%%` to it
3. Press save
4. Refresh page
5. String has reverted

**Additional information**
PrestaShop version: 1.7

Bugfix PR: #16359

### PR DESCRIPTION
**Pull request by jnvsor**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Fixes a bug preventing the saving of translations containing a `%%` literal
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #16421 
| How to test?  | Attempt a translation containing a literal `%%` - it will not be saved (Despite a green popup)

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16359)

### CODE REVIEW

**Comment by Progi1984** [comment]
@jnvsor Thank you for your contribution. Could you open an issue for describing the bug ? Thanks

**Comment by jnvsor** [comment]
@Progi1984 Added a bug report

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Hi and sorry for the delay,
why not updating the ̀regexSprintfParams` property?

**Comment by jnvsor** [comment]
Because you still want `isSprintfString` to return true if the only argument is a `%%`, otherwise you might break existing strings

**Comment by PierreRambaud** [comment]
@jnvsor This could be done with a single regex :thinking:

**Comment by jnvsor** [comment]
> This could be done with a single regex

By using capture groups? I presume there was a performance reason not to, given that someone went to the effort of making the groups in this regex non-capturing in the first place.

The options as I see them are:

1. Have one regex that matches `%%` and ignore bug #16421
2. Have one regex that doesn't match `%%` and break existing translation strings with `%%` in them (Though given the existence of #16421 there may not be many of these)
3. Have two regexes
4. Use capture groups to distinguish between matches and argument counts (Would need more testing if we went that way)

**Comment by PierreRambaud** [comment]
@jnvsor We should be able to use each characters we want :thinking: 
Having two regexes = two to maintain.
let's continue the discussion in the issue :)

**Comment by matks** [comment]
@PierreRambaud Any update :) ?

**Comment by PierreRambaud** [comment]
> @PierreRambaud Any update :) ?

Waiting for a feedback, this solution is not clean IMHO

**Comment by matks** [comment]
Hello, it seems this PR has no activity since 2020.

I close this PR, following our [review guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/) "Pull Requests may be closed after 30 days of inactivity".

If however you are still interested in it, please ping us 😉  we can reopen.

Thank you for your contribution.

**Pull request closed without merging**
